const errorHandler = (err) => {
    console.log('err.message:', err.message, 'err.code:', err.code)
    let errors = { email: '', password: '', else: ''}
    // Validating unique user registration
    // This check comes first, as if the email is already in use, there is no need to validate it at first place!
    if (err.code === 11000) {
        errors.email = 'Email already registered.'
        return errors
    }
    // Validating inputs errors 
    if (err.message.includes('User validation failed')) {
        // --- TODO ---
        // Not sure this function works as intended !! I suspect the fields are not properly rendered as errors properties.
        Object.values(err.errors).forEach(({ properties }) => {
            errors[properties.path] = properties.message
        })
    } else {
        errors.else = err.message
    }
    return errors
}

module.exports = errorHandler